src/ApparentHorizons/FastFlow.cpp

```diff
@@ -0,0 +1,302 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "ApparentHorizons/FastFlow.hpp"
+
+#include <algorithm>
+#include <cmath>
+#include <pup.h>
+#include <utility>
+
+#include "ApparentHorizons/SpherepackIterator.hpp"
+#include "ApparentHorizons/Strahlkorper.hpp"
+#include "ApparentHorizons/StrahlkorperDataBox.hpp"
+#include "ApparentHorizons/StrahlkorperGr.hpp"
+#include "ApparentHorizons/YlmSpherepack.hpp"
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/EagerMath/Magnitude.hpp"
+#include "ErrorHandling/Error.hpp"
+#include "Parallel/Abort.hpp"
+#include "PointwiseFunctions/GeneralRelativity/IndexManipulation.hpp"
+
+FastFlow::FastFlow(FastFlow::Flow::type flow, FastFlow::Alpha::type alpha,
+                   FastFlow::Beta::type beta, FastFlow::AbsTol::type abs_tol,
+                   FastFlow::TruncationTol::type trunc_tol,
+                   FastFlow::DivergenceTol::type divergence_tol,
+                   FastFlow::DivergenceIter::type divergence_iter,
+                   FastFlow::MaxIts::type max_its, const OptionContext& context)
+    : alpha_(alpha),
+      beta_(beta),
+      abs_tol_(abs_tol),
+      trunc_tol_(trunc_tol),
+      divergence_tol_(divergence_tol),
+      divergence_iter_(divergence_iter),
+      max_its_(max_its),
+      // clang-tidy std::move on trivially copyable type
+      flow_(std::move(flow)),  // NOLINT
+      current_iter_(0),
+      previous_residual_mesh_norm_(0.0),
+      min_residual_mesh_norm_(std::numeric_limits<double>::max()),
+      iter_at_min_residual_mesh_norm_(0) {
+  if (divergence_tol_ < 1.0) {
```

Comment:
give DivergenceTol a lower_bound, and then you can remove this check, noexcept the constructor, remove the option context argument